Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -69,6 +69,16 @@ open class BasicFormat: SyntaxRewriter {
     super.init(viewMode: viewMode)
   }
 
+  /// Clears all stateful data from this `BasicFormat`.
+  ///
+  /// This needs to be called between multiple `rewrite` calls to a syntax tree.
```

Comment:
Could we just override `rewrite` and do this in there instead?

Reply:
`rewrite` currently isn’t `open`. And I don’t really want to make it `open` just for that reason… Thinking about this, I thought about introducing `BasicFormat.format(_ node: some SyntaxProtocol)` but that would be the same as `SyntaxProtocol.formatted(using: BasicFormat)`. So, maybe we should just make `reset` internal and tell people to use `SyntaxProtocol.formatted`?

Reply:
Updated it to make `reset` internal again with the intention that you format code by calling `SyntaxProtocol.formatted`.